File: mRemoteV1/Connection/Connection.Icon.vb
Imports System.ComponentModel
Imports System.IO
Imports mRemote3G.App
Imports mRemote3G.Messages

Namespace Connection
    Public Class Icon
        Inherits StringConverter

        Public Shared Icons As String() = New String() {}

        Public Overloads Overrides Function GetStandardValues(context As ITypeDescriptorContext) _
            As StandardValuesCollection
            Return New StandardValuesCollection(Icons)
        End Function

        Public Overloads Overrides Function GetStandardValuesExclusive(context As ITypeDescriptorContext) As Boolean
            Return True
        End Function

        Public Overloads Overrides Function GetStandardValuesSupported(context As ITypeDescriptorContext) As Boolean
            Return True
        End Function

        Public Shared Function FromString(IconName As String) As Drawing.Icon
            Try
                Dim IconPath As String = My.Application.Info.DirectoryPath & "\Icons\" & IconName & ".ico"

                If File.Exists(IconPath) Then
                    Dim nI As New Drawing.Icon(IconPath)

                    Return nI
                End If
            Catch ex As Exception
                Runtime.MessageCollector.AddMessage(MessageClass.ErrorMsg,
                                                    "Couldn't get Icon from String" & vbNewLine & ex.ToString())
            End Try

            Return Nothing
        End Function
    End Class
End Namespace

File: mRemoteV1/Connection/ConnectionIcon.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.IO;
using mRemoteNG.App;
using mRemoteNG.App.Info;
using mRemoteNG.Messages;

namespace mRemoteNG.Connection
{
    public class ConnectionIcon : StringConverter
    {
        public static string[] Icons = {};

        public override StandardValuesCollection GetStandardValues(ITypeDescriptorContext context)
        {
            return new StandardValuesCollection(Icons);
        }

        public override bool GetStandardValuesExclusive(ITypeDescriptorContext context)
        {
            return true;
        }

        public override bool GetStandardValuesSupported(ITypeDescriptorContext context)
        {
            return true;
        }

        public static Icon FromString(string iconName)
        {
            try
            {
                var iconPath = $"{GeneralAppInfo.HomePath}\\Icons\\{iconName}.ico";

                if (File.Exists(iconPath))
                {
                    var nI = new Icon(iconPath);
                    return nI;
                }
            }
            catch (Exception ex)
            {
                Runtime.MessageCollector.AddMessage(MessageClass.ErrorMsg,
                    $"Couldn\'t get Icon from String" + Environment.NewLine + ex.Message);
            }
            return null;
        }
    }
}